tests/test_main.cpp

```diff
@@ -203,7 +203,14 @@ struct CataListener : Catch::TestEventListenerBase {
 
     void testRunStarting( Catch::TestRunInfo const & ) override {
         if( needs_game ) {
-            init_global_game_state( mods, option_overrides_for_test_suite, user_dir );
+            try {
+                init_global_game_state( mods, option_overrides_for_test_suite, user_dir );
+            } catch( ... ) {
+                DebugLog( D_INFO, DC_ALL ) << "Error during initializing global game state" << std::endl;
+                // NOLINTNEXTLINE(cata-tests-must-restore-global-state)
+                error_during_initialization = true;
+                throw;
+            }
             // NOLINTNEXTLINE(cata-tests-must-restore-global-state)
             error_during_initialization = debug_has_error_been_observed();
```

Comment:
`error_during_initialization` is also set when there are debug messages, but in that case the save should probably still be saved if loading succeeds.

Reply:
I changed it to a dedicated variable to mark initialisation failure.